src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/ErrorType.cs

```diff
@@ -18,17 +18,6 @@ internal sealed class ErrorType : CType
         public Name nameText;
         public TypeArray typeArgs;
 
-        public bool HasParent() { return _pParentType != null || _pParentNS != null; }
-
-        public bool HasTypeParent() { return _pParentType != null; }
-        public CType GetTypeParent() { return _pParentType; }
-        public void SetTypeParent(CType pType) { _pParentType = pType; }
-
-        public bool HasNSParent() { return _pParentNS != null; }
-        public AssemblyQualifiedNamespaceSymbol GetNSParent() { return _pParentNS; }
-        public void SetNSParent(AssemblyQualifiedNamespaceSymbol pNS) { _pParentNS = pNS; }
-
-        private CType _pParentType;
-        private AssemblyQualifiedNamespaceSymbol _pParentNS;
+        public bool HasParent => nameText != null;
     }
```

Comment:
I suggest putting some description here. This is not obvious from reading the code.
Also, since you're refactoring most usages of this type, it might be worth it to convert it to a C# class to better guard for future bugs (constructor, private setters for properties), and documentation on why nameText indicates whether it has a parent or not.